### DISCUSSION TITLE
ccsid difference on 2

### DISCUSSION THREAD

**Discussion by JeanLand400** [codefori/vscode-ibmi#1372] [General]
Hi,

I am a beginner in IBM i , and I do love to use Code for IBMi extension to do most of my work .

Recently it came to my eye I had an issue whenever saving source code files or copying from one source to another .
It so happens be fine whenever I work on a ccsid 297 source file ( you guessed right , I'm french ) , but incorrectly ( special char £!@ are converted ) when the source file is of ccsid 37 .
The config is as follow : 
QCCSID 65535 ,
QCHRID 695,
QCHRIDCTL *DEVD,
QLANGID FRA,
QCNTRYID FR .

In the extension I have tried to change sourceFileCCSID but with no success, also, I am not able to change my user profile settings ( which are mostly *sysval) .

Is the issue my poorly configured extension , or does it lies elsewhere ?

**Comment by chrjorgensen** [comment]
Try setting the CCSID value on your user profile to 297, and report back.

**Reply by JeanLand400** [reply]
It made things worse , since there is a conversion on both source files now .
edit : also it converted my special characters to hyeroglyphs on both files when reverting the change.

**Reply by worksofliam** [reply]
The big thing at play here is QCCSID 65535. SQL doesn't work very well. If you have SQL disabled, then I imagine there should be no issue with converting the encodings. If you have SQL enabled, but QCCSID 65535, ensure you've checked out our [CCSID](https://halcyon-tech.github.io/docs/#/pages/tips/ccsid) docs and make sure your system is setup correctly.

**Reply by JeanLand400** [reply]
I have SQL disabled for now , should I enable it ?

**Reply by chrjorgensen** [reply]
@JeanLand400 Did you make any progress on this issue? My advice would be to set CCSID on your userprofile (to 297) and enable SQL. Also check you SSH server setup according to the [docs](https://halcyon-tech.github.io/docs/#/pages/tips/ccsid).
Btw, I don't think it will work setting the `sourceFileCCSID` value in the extension, since you have source files in different CCSID's.
